packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceRetrieveMetadata/forceSourceRetrieveCmp.test.ts

```diff
@@ -153,8 +153,8 @@ describe('Force Source Retrieve Component(s)', () => {
           new SourceComponent({
             name: 'MyClassB',
             type,
-            content: path.join(type.directoryName, 'MyClassB.cls'),
-            xml: path.join(type.directoryName, 'MyClassB.cls-meta.xml')
+            content: path.join(String(type.directoryName), 'MyClassB.cls'),
```

Comment:
`String()` here doesn't seem to be needed.  Also, in general it's probably best to avoid using `String()` since there's always the confusion over whether it's a TypeScript `String` or a JavaScript `String`, and when it's a TypeScript `String`, it's a class object and String !== string.

Reply:
TypeScript is beefing about it if I don't make the change.  What is the best way to resolve this TS error?

Reply:
Weird, I must have had a cached version of something.  I wasn't getting any TS errors, and previously `MetadataType.directoryName` was defined as a string.

I just did a git clean, install, and then compile, and now it's defined as `directoryName?: string`
To get past this, use the `!` operator...
```
content: path.join(type.directoryName!, 'MyClassB.cls'),
```
...using `!` is preferred over using `String()`. because `String` has different meaning in TS vs JS, and b/c calling `String()` is calling the `String` class constructor.

Reply:
👍🏽 updated to use the non-null assertion operator in latest commit.

Reply:
I was re-visiting this change and decided to revert back to using String() for a few reasons:

* In the upgraded SDR library, `directoryName` is now optional as you mentioned above.  The non-null assertion operator tells TypeScript that VSCE _knows_ that this value is not null, when it could be.  
* String() is a node function that [coerces the value to a string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#string_primitives_and_string_objects) (`new String()` is the pattern that creates a String object and could cause comparison issues and should generally be avoided)
* Using String() in this scenario is an [established pattern in the VSCE code base](https://github.com/forcedotcom/salesforcedx-vscode/pull/4348#discussion_r938251211)

@jeffb-sfdc let me know if you'd like to discuss.  For these reasons I'd prefer `String()` over using the `!`.